doc/topics/blocking-vs-non-blocking.md

```diff
@@ -0,0 +1,140 @@
+# Overview of Blocking vs Non-Blocking
+
+This overview covers the difference between blocking and non-blocking calls in
+Node.js. This overview will refer to the event loop and libuv but no prior
+knowledge of those topics is required. Readers are assumed to have a basic
+understanding of the JavaScript language and Node.js callback pattern.
+
+> In this overview, "I/O" refers primarily to interaction with the system's disk
+and network supported by [libuv](http://libuv.org/).
+
+
+## Blocking
+
+Blocking is when the execution of additional JavaScript in the Node.js process
+must wait until an I/O operation completes. Blocking may occur when using any of
+the synchronous I/O methods in the Node.js standard library that use libuv. Use
+of blocking methods prevents the event loop from doing additional work while
+waiting for I/O to complete.
+
+In Node.js, JavaScript that exhibits poor performance due to being CPU intensive
+rather than I/O bound, isn't typically referred to as blocking.
```

Comment:
There are two notions of what "blocking" is - one is "blocking the event loop" and one is "blocked as in the [waiting thread state](http://www.macdesign.net/capella/it4813/images/Process_states_in_UNIX-color.png)".

When people use "blocking" in node sometimes they mean one and sometimes they mean the other.